06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js

```diff
@@ -0,0 +1,345 @@
+'use strict';
+
+const InvertedIndex = require('../public/builds/InvertedIndex').InvertedIndex;
+
+const books = require('../booksamples/books.json');
+const booksInvalidContent = require('../booksamples/booksInvalidContent.json');
+const bookBadJson = require('../booksamples/bookBadJson.json');
+const bookEmptyJson = require('../booksamples/bookEmptyJson.json');
+
+const bookText = ' ';
+
+const iDex = new InvertedIndex();
+
+describe('iDex Inverted Index', () => {
+  beforeEach((done) => {
+    iDex.createIndex('books', books);
+    iDex.createIndex('booksInvalidContent', booksInvalidContent);
+    iDex.createIndex('bookBadJson', bookBadJson);
+    iDex.createIndex('bookText', bookText);
+    done();
+  });
+  describe('The InvertedIndex class', () => {
+    it('should be defined', () => {
+      expect(InvertedIndex).toBeDefined();
+    });
+
+    it('should have a defined instance', () => {
+      expect(iDex).toBeDefined();
+      expect(iDex instanceof InvertedIndex).toBeTruthy();
+    });
+
+    it('should default instance properties to object types', () => {
+      expect(iDex.iDexMapper).toBeDefined();
+      expect(typeof iDex.iDexMapper).toEqual('object');
+
+      expect(iDex.numberOfDocuments).toBeDefined();
+      expect(typeof iDex.numberOfDocuments).toEqual('object');
+
+      expect(iDex.unIndexedBooks).toBeDefined();
+      expect(iDex.unIndexedBooks).toEqual({});
+      expect(typeof iDex.unIndexedBooks).toEqual('object');
+    });
+  });
+
+  describe('The readFile method', () => {
+    it('should return the json file if it is valid', () => {
+      expect(InvertedIndex.readFile(JSON.stringify(books))).toEqual((books));
+    });
+
+    it('should return the json file for an empty json', () => {
+      expect(InvertedIndex.readFile(JSON.stringify(bookEmptyJson)))
+        .toEqual((bookEmptyJson));
+    });
+
+    it('should return false if the file cannot be parsed', () => {
+      expect(InvertedIndex.readFile((bookText))).toBe((false));
+      expect(InvertedIndex.readFile((bookText))).toBeFalsy();
+    });
+  });
```

Comment:
Because, I wrote the readfile to just parse a json string to a JS object if it is a valid json or return false if it is unable to do so.

I the condition makes it seem like a validation, which may still be the case.